pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanVisitor.java

```diff
@@ -137,7 +140,21 @@ public Void visitSort(SortNode node, DispatchablePlanContext context) {
   @Override
   public Void visitTableScan(TableScanNode node, DispatchablePlanContext context) {
     DispatchablePlanMetadata dispatchablePlanMetadata = getOrCreateDispatchablePlanMetadata(node, context);
-    dispatchablePlanMetadata.addScannedTable(_tableCache.getActualTableName(node.getTableName()));
+
+    LogicalTableRouteInfo logicalTableRouteInfo = null;
+
+    String tableNameInNode = node.getTableName();
+    String tableName = _tableCache.getActualTableName(tableNameInNode);
+    if (tableName == null) {
+      tableName = _tableCache.getActualLogicalTableName(tableNameInNode);
+      Preconditions.checkNotNull(tableName, "Logical table config not found in table cache: " + tableName);
```

Comment:
The condition message should reference `tableNameInNode` and not `tableName` as would be null.